Handle setpoint

olesot opened the pull request:
The point with this branch is to gain better control over the setpoints of the controller. 

* Changed the controllers from updating all setpoints to updating only the state under control
* Add new functionalities to setpoint and state class
* Remove oblete code
* Publishing Debug message (state and setpoint)

Needs reviews and polishing.

olesot committed:
Fixes #61

olesot committed:
fix roslint

olesot committed:
Merge branch 'master' into handle_setpoint

olesot committed:
Add function to get orientation in euler angles

sindrehan commented:
[CodeFactor](https://www.codefactor.io/repository/github/vortexntnu/rov-control) found an issue: Duplicate Code

It's currently on:
[vortex_controller\src\state.cpp:54-67
](https://github.com/vortexntnu/rov-control/blob/4c5bcc93122c7f9ff64d7e0c6cd5619c79c28867/vortex_controller/src/state.cpp#L54-L67)[vortex_controller\src\setpoints.cpp:101-114
](https://github.com/vortexntnu/rov-control/blob/4c5bcc93122c7f9ff64d7e0c6cd5619c79c28867/vortex_controller/src/setpoints.cpp#L101-L114)Commit 4c5bcc93122c7f9ff64d7e0c6cd5619c79c28867

olesot committed:
Merge remote-tracking branch 'origin/master' into handle_setpoint

olesot committed:
Merge updatePositionSetpoint and updateOrientationSetpoint

updateSetpoint is updating the setpoint axis specified as parameter.

olesot committed:
Remove outdated variables and functionality

pose_rate and time is not used in setpoint and is removed. The update function is changed to only updating wrench.

olesot committed:
Fix euler angle indexing from x to z

+ fix roslint

olesot commented:
The duplicate code check fails. 
As I see it I can implement the duplicated code in controller_ros, but not sure if it makes more sense to do that than keep current implementation. Any other suggestions?
